### PR TITLE
Improved test suite for self heal feature

### PR DESCRIPTION
**Pull request by vpandey-RH**
Improved test suite for heal process.

Signed-off-by: Vishal Pandey <vpandey@redhat.com>

### CODE REVIEW

**Comment by aravindavk** [comment]
add to whitelist

File: e2e/glustershd_test.go [hunk 1]
```diff
@@ -47,12 +54,55 @@ func testSelfHeal(t *testing.T, tc *testCluster) {
...
+		if getBricksStatus[brick].Info.PeerID.String() == tc.gds[0].PeerID() {
+			process, err := os.FindProcess(getBricksStatus[brick].Pid)
+			r.Nil(err, fmt.Sprintf("failed to find bricks pid: %s", err))
+			err = process.Signal(syscall.Signal(15))
+			r.Nil(err, fmt.Sprintf("failed to kill bricks: %s", err))
+			break
+		}
+	}
+
+	f1, err := os.OpenFile(mntPath+"/file1.txt", os.O_RDWR, 0777)
```
Comment (Madhu-1) [inline e2e/glustershd_test.go]: why permission is `0777`?

File: e2e/glustershd_test.go [hunk 2]
```diff
@@ -47,12 +54,55 @@ func testSelfHeal(t *testing.T, tc *testCluster) {
...
+	f, err := os.Create(mntPath + "/file1.txt")
+	r.Nil(err, fmt.Sprintf("file creation failed: %s", err))
+	f.Close()
+
+	getBricksStatus, err := client.BricksStatus(volname)
+	r.Nil(err, fmt.Sprintf("brick status operation failed: %s", err))
+	for brick := range getBricksStatus {
+		if getBricksStatus[brick].Info.PeerID.String() == tc.gds[0].PeerID() {
+			process, err := os.FindProcess(getBricksStatus[brick].Pid)
+			r.Nil(err, fmt.Sprintf("failed to find bricks pid: %s", err))
```
Comment (aravindavk) [inline e2e/glustershd_test.go]: This test will be silently ignored if condition in line 75 is not satisfied. Add a test before for loop to check required number of BricksStatus with gds[0].PeerID

File: e2e/glustershd_test.go [hunk 3]
```diff
@@ -47,12 +54,55 @@ func testSelfHeal(t *testing.T, tc *testCluster) {
...
-	_, err = client.SelfHealInfo(vol1.Name, "info-summary")
-	r.Nil(err)
 	_, err = client.SelfHealInfo(vol1.Name, "split-brain-info")
 	r.Nil(err)
+	healInfo, err := client.SelfHealInfo(vol1.Name, "info-summary")
+	r.Nil(err)
+
+	for node := range healInfo {
+		if healInfo[node].Status == "Connected" {
+			r.NotNil(checkForPendingHeals(&healInfo[node]))
```
Comment (aravindavk) [inline e2e/glustershd_test.go]: Same here. Add a test to see required number of nodes are in connected state

File: e2e/glustershd_test.go [hunk 4]
```diff
@@ -63,6 +113,15 @@ func testSelfHeal(t *testing.T, tc *testCluster) {
 	r.True(isProcessRunning(pidpath), "glustershd is not running")
 
 	r.Nil(client.SelfHeal(vol1.Name, "index"))
+
+	healInfo, err = client.SelfHealInfo(vol1.Name, "info-summary")
+	r.Nil(err)
+	for node := range healInfo {
+		if healInfo[node].Status == "Connected" {
+			r.Nil(checkForPendingHeals(&healInfo[node]))
```
Comment (aravindavk) [inline e2e/glustershd_test.go]: same here

**Pull request merged**